src/main/java/user/database/UserFileReader.java

```diff
@@ -0,0 +1,26 @@
+package user.database;
+
+import user.entities.User;
+
+import java.io.*;
+import java.util.HashMap;
+import java.util.Map;
+
+public class UserFileReader {
+
+    static public Map<String, User> getUserMap(String filename){
+        try {
+        InputStream inputStream = new FileInputStream(filename);
+        InputStream buffer = new BufferedInputStream(inputStream);
+        ObjectInput input = new ObjectInputStream(buffer);
+
+        Map<String, User> UserMap= (Map<String, User>) input.readObject();
```

Comment:
Well, I don't know why casting to a map is considered unsafe, while casting to a UserList is considered safe. Since HashMap is also implemented as serializable, using HashMap or UserList should be the same ...

Reply:
Since I'm pretty confident the code is correct, it's safe to use @SuppressWarnings("unchecked") to close the warning.

Reply:
I have no idea why but it seems to them it makes some differences TT

![image](https://user-images.githubusercontent.com/43018428/201228350-92cad772-e628-4ad9-9fc0-f6ec3e58439c.png)

Reply:
Sure I'll do that, thx!

> Since I'm pretty confident the code is correct, it's safe to use @SuppressWarnings("unchecked") to close the warning.

Reply:
I guess it means the elements inside the Map are unchecked. Like the String and User. But since the only portal to write into the database is DatabaseWriter, and the writer will check the input data, so it should be safe enough.